FF1Lib/asm/1E_B100_DamageTilesKill.asm

```diff
@@ -35,20 +40,60 @@ Loop:
   LDA ch_ailments, X
   CMP #$01
   BEQ AlreadyDead
-  
-  LDA ch_curhp+1, X     ; check high byte of HP
-  BNE DmgSubtract       ; if nonzero (> 255 HP), deal this guy damage
 
-  LDA ch_curhp, X       ; otherwise, check low byte
-  CMP #2                ; if >=2, just substract hp
-  BCS DmgSubtract
+	LDA #$01				; only do any of the special armor checks if the randomizer has set this to 0
+	BNE AssignDamage
+
+	@VolcanoCheck:
+		LDA cur_map				; get the current map
+		CMP #$0E				; see if it's Volcano
+		BEQ @Volcano
+		CMP #$21
+		BEQ @Volcano
+		CMP #$22
+		BEQ @Volcano
+		CMP #$23
+		BEQ @Volcano
+		CMP #$24
+		BEQ @Volcano
+		JMP @IceCheck
+
+	@Volcano:
+		LDA #$08 ; Ice Armor ID
```

Comment:
You could make your check code simpler by looking for an equipped Ice armor #$88 instead of checking all equipped armor pieces to be an ice armor.

Reply:
Sure, that makes sense.